Source/Timeline/TimelineView.swift

```diff
@@ -23,21 +27,62 @@ public class TimelineView: UIView, ReusableView {
   }
 
   var eventViews = [EventView]()
-  public var layoutAttributes = [EventLayoutAttributes]() {
-    didSet {
+  public private(set) var regularLayoutAttributes = [EventLayoutAttributes]()
+  public private(set) var allDayLayoutAttributes = [EventLayoutAttributes]()
+  
+  public var layoutAttributes: [EventLayoutAttributes] {
+    set {
+      
+      // update layout attributes by separating allday from non all day events
+      allDayLayoutAttributes.removeAll()
+      regularLayoutAttributes.removeAll()
+      for anEventLayoutAttribute in newValue {
+        let eventDescriptor = anEventLayoutAttribute.descriptor
+        if eventDescriptor.isAllDay {
+          allDayLayoutAttributes.append(anEventLayoutAttribute)
+        } else {
+          regularLayoutAttributes.append(anEventLayoutAttribute)
+        }
+      }
+      
       recalculateEventLayout()
       prepareEventViews()
+      allDayView.events = allDayLayoutAttributes.map { $0.descriptor }
+      
+      //TODO: resize to height of zero if count is zero vs using hidden
```

Comment:
Hidden is perfectly fine, since it hide the view completely.

Reply:
sounds good!

Reply:
It's even was mentioned at this year's WWDC: https://developer.apple.com/videos/play/wwdc2018/220/